app/code/Magento/AsynchronousOperations/Model/Operation.php

```diff
@@ -162,6 +163,23 @@ public function setErrorCode($errorCode)
         return $this->setData(self::ERROR_CODE, $errorCode);
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function getOperationKey(): ?int
+    {
+        return $this->getData(self::OPERATION_KEY);
```

Comment:
AFAIK when data returning from DB - it might be a string. Could you convert it to int here?